### PR TITLE
Update TCK for EL 5.0

### PR DESCRIPTION
**Pull request by markt-asf**
---
name: Pull Request
about: Create a pull request for a Platform TCK change
title: 'Update TCK for EL 5.0'
labels: ''
assignees: ''

---
**Fixes Issue**
#749 

**Related Issue(s)**
#740

**Describe the change**
Updates existing tests and adds new tests for EL 5.0

**Additional context**
This PR is currently in draft as it is dependent on a number of changes before it can be applied
- The TCK needs to be compiled with Java 11 and with Java 11 as the minimum version (#740)
- Glassfish needs to update to at least EL 5.0.0-M1 to pick up the API changes under test
- The tests added for MethodReference need to be confirmed as valid. I think they are correct but I can't check them until the previous two dependencies have been met

Additionally, this is by far the largest change I have made to a TCK. I suspect I may not have followed the expected approach in some cases and some refactoring will be required.

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @scottmarlow

### CODE REVIEW

**Comment by Thihup** [comment]
@markt-asf Would it be possible to add some TCK tests related to Map literal? I think this part of the spec, the TCK doesn't cover it very well.

https://github.com/eclipse-ee4j/el-ri/issues/161

**Review by gurunrao — COMMENTED** [review]
update copyright to 2021.

**Comment by markt-asf** [comment]
Thanks for the JDK 11 updates. I am able to build the EL TCK against a local build of the EL API now.

It appears I have a couple of issues to fix in the original draft as well as the comments above to address. I'll push an updated version of the PR when I have completed that work.

File: src/com/sun/ts/tests/el/api/jakarta_el/methodexpression/ELClient.java [hunk 1]
```diff
@@ -419,4 +422,227 @@ public void methodExpressionSerializableTest() throws Fault {
...
+      MethodsBean bean = new MethodsBean();
+      ValueExpression ve = expFactory.createValueExpression(bean, MethodsBean.class);
+      context.getVariableMapper().setVariable("bean", ve);
+
+      MethodExpression me = expFactory.createMethodExpression(context, exprStr, String.class, null);
+      if (!ExpressionTest.testMethodExpression(me, context, exprStr, null, "CharSequence", false, buf)) {
+        pass = false;
+      }
+    } catch (Exception ex) {
+      throw new Fault(ex);
```
[the remark this answers is not in the document]
Comment (markt-asf) [reply src/com/sun/ts/tests/el/api/jakarta_el/methodexpression/ELClient.java]: Done.

File: src/com/sun/ts/tests/el/api/jakarta_el/methodexpression/ELClient.java [hunk 2]
```diff
@@ -419,4 +422,227 @@ public void methodExpressionSerializableTest() throws Fault {
     else
       TestUtil.logTrace(buf.toString());
   }
+
+  /*
+   * @testName: methodExpressionMatchingExactPreferredTest
+   *
```
[the remark this answers is not in the document]
Comment (markt-asf) [reply src/com/sun/ts/tests/el/api/jakarta_el/methodexpression/ELClient.java:428]: Done.

**Comment by markt-asf** [comment]
This updated PR addresses all the review comments and includes the requests additional TCK tests for Map Literals. It also includes similar tests for Set and List literals. Thanks for the suggestion - I found and fixed a bug in Tomcat's EL implementation as a result.
I'm leaving this PR as draft as there is one final EL issue to resolve. I hope to get to that today and once the API changes are complete I'll wrap up this PR and move it out of draft.

**Comment by markt-asf** [comment]
This PR now contains all the necessary changes for the changes in EL 5.0

**Review by scottmarlow — APPROVED** [review]

**Comment by markt-asf** [comment]
Just a rebase. No other changes.

**Comment by markt-asf** [comment]
This should now be complete for EL 5.0

File: src/com/sun/ts/tests/common/el/api/expression/ExpressionTest.java [hunk 3]
```diff
@@ -87,6 +89,85 @@ public static boolean testMethodInfo(MethodInfo minfo, String expectedName,
...
+      Object[] expectedParamValues, StringBuffer buf) {
+
+    boolean pass = true;
+    
+    Object base = mref.getBase();
+    MethodInfo minfo = mref.getMethodInfo();
+    Annotation[] annotations = mref.getAnnotations();
+    Object[] parameterValues = mref.getEvaluatedParameters();
+    
+    if (base == null) {
```
[the remark this answers is not in the document]
Comment (scottmarlow) [reply src/com/sun/ts/tests/common/el/api/expression/ExpressionTest.java:104]: @gurunrao is this change blocking your approval or are there other pending changes your waiting on to approve?

[the remark this answers is not in the document]
Comment (gurunrao) [reply src/com/sun/ts/tests/common/el/api/expression/ExpressionTest.java:104]: > @gurunrao is this change blocking your approval or are there other pending changes your waiting on to approve?

ya, was waiting for that. approved anyway.

**Review by gurunrao — APPROVED** [review]

**Comment by gurunrao** [comment]
> This should now be complete for EL 5.0

@markt-asf - Please resolve conflicts at `"src/com/sun/ts/tests/signaturetest/signature-repository/jakarta.el.sig_4.0_se8"` for the merge to proceed.

**Comment by markt-asf** [comment]
Force push should have resolved the conflicts.

**Pull request merged**